src/types/params.ts

```diff
@@ -46,7 +45,7 @@ export function isParamGetSet(value: Param): value is ParamGetSet {
  * @returns The extracted parameter name, or never if the parameter string is empty.
  */
 export type ExtractParamName<
-  TParam extends string
+  TParam extends PropertyKey
```

Comment:
this just makes it more convenient to use the utility, doesn't change functionality